h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -406,6 +391,32 @@ private void gridSearch(Grid<MP> grid) {
     }
   }
 
+  /**
+   * see {@code RandomDiscreteValueSearchCriteria.max_runtime_secs} for reconciliation logic
+   */
+  private void reconcileMaxRuntime(Key<Grid<MP>> gridKey, HyperSpaceWalker.HyperSpaceIterator<? extends Model.Parameters> it, Model.Parameters params) {
+    double grid_max_runtime_secs = it.max_runtime_secs();
+
+    double time_remaining_secs = Double.MAX_VALUE;
+    if (grid_max_runtime_secs > 0) {
+      time_remaining_secs = it.time_remaining_secs();
+      Log.info("Grid time is limited to: " + grid_max_runtime_secs + " for grid: " + gridKey + ". Remaining time is: " + time_remaining_secs);
+      if (time_remaining_secs < 0) {
+        Log.info("Grid max_runtime_secs of " + grid_max_runtime_secs + " secs has expired; stopping early.");
+        throw new Job.JobCancelledException();
+      }
+    }
+
+    if (params._max_runtime_secs > 0 ) {
+      double was = params._max_runtime_secs;
+      params._max_runtime_secs = Math.min(params._max_runtime_secs, time_remaining_secs);
+      Log.info("Due to the grid time limit, changing model max runtime from: " + was + " secs to: " + params._max_runtime_secs + " secs.");
+    } else { // params._max_runtime_secs == 0
+      params._max_runtime_secs = time_remaining_secs;
+      Log.info("Due to the grid time limit, changing model max runtime to: " + params._max_runtime_secs + " secs.");
+    }
+  }
+
```

Comment:
re-re-reviewed and approved.

legacy `if (max_runtime_secs > 0) {` before checking the `params._max_runtime_secs` was preventing a correct application of the rule, thanks for the fix.
Must the reason why the test above was passing when it shouldn't have.

Reply:
@seb-h2o 
> Must be the reason why the test above was passing when it shouldn't have

i think yes